content/03.methods.md

```diff
@@ -1,17 +1,99 @@
+Nonetheless, our [results](#sec:results) show that this is not a main problem and the model still produces high-quality outcomes.
+Additionally, since the goal of our tool is to revise a paragraph, by default we set the maximum number of tokens (parameter `max_tokens`) as twice the estimated number of tokens in the paragraph (one token approximately represents four characters, as indicated in [@url:https://beta.openai.com/tokenizer]).
+The tool automatically adjusts this parameter and performs the request again if a related error is returned by the API.
+The user can force the tool to either use a fixed value for `max_tokens` for all paragraphs, or change the fraction of maximum tokens based on the estimated paragraph size (two by default).
+
+
+The language models used are stochastic: they will generate a different revision for the same input paragraph each time.
+This behavior can be changed by using the "sampling temperature" or "nucleus sampling" parameters (we use `temperature=0.5` by default).
+Although we selected default values that worked well in different scenarios, these parameters can changed by the user if necessary to make the model more deterministic.
+The user can also instruct the model to generate, for each paragraph, several completions and select the one with the highest log probability per token, what can improve the quality of the revision.
+Our implementation generates only one completion (parameter `best_of=1`) to avoid potentially excessive costs for the user, but this behavior can also be easily changed.
+Additionally, our workflow allows to process either the entire manuscript or individual sections.
+This allows to control costs more effectively while focusing on a single piece of text in which the user can run the tool several times and pick the prefered revised text.
+
+
+Some paragraphs are simpler to process than others.
+For instance, the abstract is a set of sentences with no citations, whereas a paragraph from the Introduction section has several references to other scientific papers.
+A paragraph in the Results section has less citations but many references to figures or tables, where enough details about the experiments must be provided to understand and interpret the outcomes.
+The Method section is more dependent on the type of paper, but in general it has to provide technical details and sometimes mathematical formulas and equations.
+We designed some prompts that are specific to each section (Figure {@fig:ai_revision}b).
+Figures and tables captions, as well as paragraphs that contain only one or two sentences and less than sixty words, are not processed and copied directly to the output file.
+When each paragraph is being processed, our tool smoothly handles common API errors (such as "sever overloaded") and retries to revise up to five times by default, although this can also be configured by the user.
```

Comment:
Yea - this definitely repeats what is above.

Reply:
Ops. It's fixed now.